internal/pkg/aws/cloudwatch/cloudwatch.go

```diff
@@ -49,38 +59,57 @@ type AlarmStatus struct {
 
 // New returns a CloudWatch struct configured against the input session.
 func New(s *session.Session) *CloudWatch {
-	return &CloudWatch{
-		cwClient: cloudwatch.New(s),
-		rgClient: rg.New(s),
+	cw := &CloudWatch{
+		client: cloudwatch.New(s),
+	}
+	cw.initRgClient = func() {
+		cw.rgClient = rg.New(s)
 	}
+	cw.initAssclient = func() {
+		cw.assClient = aas.New(s)
+	}
+	return cw
 }
 
-// GetAlarmsWithTags returns all the CloudWatch alarms that have the resource tags.
-func (cw *CloudWatch) GetAlarmsWithTags(tags map[string]string) ([]AlarmStatus, error) {
-	var alarmNames []*string
+// ECSServiceAutoscalingAlarms returns the CloudWatch alarms associated with the
+// auto scaling policies attached to the ECS service.
+func (cw *CloudWatch) ECSServiceAutoscalingAlarms(cluster, service string) ([]AlarmStatus, error) {
+	cw.initAssclient()
+	alarmNames, err := cw.assClient.ECSServiceAlarmNames(cluster, service)
+	if err != nil {
+		return nil, fmt.Errorf("retrieve auto scaling alarm names for ECS service %s/%s: %w", cluster, service, err)
+	}
+	return cw.alarmStatus(alarmNames)
+}
 
+// AlarmsWithTags returns all the CloudWatch alarms that have the resource tags.
+func (cw *CloudWatch) AlarmsWithTags(tags map[string]string) ([]AlarmStatus, error) {
+	cw.initRgClient()
+	var alarmNames []string
 	resources, err := cw.rgClient.GetResourcesByTags(cloudwatchResourceType, tags)
 	if err != nil {
 		return nil, err
 	}
-
 	for _, resource := range resources {
-		name, err := cw.getAlarmName(resource.ARN)
+		name, err := getAlarmName(resource.ARN)
 		if err != nil {
 			return nil, err
 		}
-		alarmNames = append(alarmNames, name)
+		alarmNames = append(alarmNames, aws.StringValue(name))
 	}
+	return cw.alarmStatus(alarmNames)
+}
 
-	// Return an empty array since DescribeAlarms will return all alarms if "AlarmNames" is an empty array.
-	if len(alarmNames) == 0 {
+func (cw *CloudWatch) alarmStatus(alarms []string) ([]AlarmStatus, error) {
```

Comment:
What do you think of making this public and getting rid of the `ECSServiceAutoscalingAlarms` method? It brings another dependency to this pkg which is surprising a little.

Instead we can move the `ecsServiceAutoscalingAlarms` method to be on the `describe.ServiceStatus` struct.